src/com/company/Zadania.java

```diff
@@ -1332,5 +1332,29 @@ public void zadanie43() {
             System.out.print(x + ",");
         }
     }
+
+    //Write a program, which will find all such numbers between 1000 and 3000 (both included)
+    // such that each digit of the number is an even number.
+    //The numbers obtained should be printed in a comma-separated sequence on a single line.
+
+    public void zadanie44() {
+        int helper;
+        int digit;
+        int counter = 0;
+        for (int i = 1000; i < 3001; i++) {
```

Comment:
za duzo myslenia wymagasz od czytacza kodu
<= 3000

Reply:
to i tak kompilator zoptymalizuje, ale
i++ -> robi kopie, zwraca, dodaje do i jedynke
++i -> dodaje do i jedynke, zwraca referencje (nie tworzy kopii bez sensu)

wniosek: warto ++i w forze pisac - wyjedziesz na bardziej ogarnietego (jak bedziesz umial uargumentowac)